src/protocols/rdp/rdp_fs.c

```diff
@@ -607,11 +608,10 @@ const char* guac_rdp_fs_read_dir(guac_rdp_fs* fs, int file_id) {
 int guac_rdp_fs_normalize_path(const char* path, char* abs_path) {
 
     int i;
-    int path_depth = 0;
+    int path_depth = 1;
```

Comment:
What is this change about?

Reply:
I will check.

I should probably add unit tests for the path manipulation bits, too, honestly.

Reply:
> What did you find?

The additional leading `""` path component is used to ensure the final, normalized path has a leading backslash following `guac_strljoin()`. I'm adding documentation to that effect.

> Are you planning on adding more unit tests in this PR?

Yes, to verify that these various normalization functions continue to function as intended.

Reply:
OK - I've cleaned up the SFTP and RDP filesystem normalization logic and removed the need for setting `path_depth = 1`, which the unit tests I've just added revealed to be incorrect.